src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -350,9 +358,16 @@ private void TryClosingDocumentsForCookie(uint cookie)
 
                     foreach (var documentId in documentIds)
                     {
-                        if (_workspace.IsDocumentOpen(documentId) && !_workspace._documentsNotFromFiles.Contains(documentId))
+                        if (!_workspace._documentsNotFromFiles.Contains(documentId))
                         {
-                            w.OnDocumentClosed(documentId, new FileTextLoader(moniker, defaultEncoding: null));
+                            if (_workspace.IsDocumentOpen(documentId))
+                            {
+                                w.OnDocumentClosed(documentId, new FileTextLoader(moniker, defaultEncoding: null));
+                            }
+                            else if (w.CurrentSolution.ContainsAdditionalDocument(documentId))
```

Comment:
Shouldn't this be paralleling the logic in the open case above? You need to check if it's a regular document before OnDocumentClosed, and if it's not that assert it's an AdditionalDocument?

Reply:
I'll make it consistent.